torch/csrc/utils/python_strings.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <Python.h>
+#include <stdexcept>
+#include "object_ptr.h"
+
+// Utilities for handling Python strings. Note that PyString, when defined, is
+// the same as PyBytes.
+
+// Returns true if obj is a bytes/str or unicode object
+// In existing CPython implementations, this does not require the GIL
```

Comment:
I think this has been fixed by @albanD. All arguments are unpacked before giving up the GIL.